app/packages/spaces/src/components/Space.tsx

```diff
@@ -14,16 +17,53 @@ import { PanelContainer, PanelTabs, SpaceContainer } from "./StyledElements";
 export default function Space({ node, id }: SpaceProps) {
   const { spaces } = useSpaces(id);
   const autoPosition = usePanelTabAutoPosition();
+  const spaceRef = useRef<AllotmentHandle>(null);
+  const previousSizesRef = useRef<string[]>();
+  const currentTotalSize = useRef();
+  const { sizes } = node;
+
+  const setSpaceSizes = useCallback(
+    debounce((node, sizes) => {
+      currentTotalSize.current = sizes.reduce((total, item) => total + item, 0);
+      const relativeSizes = getRelativeSizes(sizes);
+      previousSizesRef.current = relativeSizes;
+      spaces.setNodeSizes(node, relativeSizes);
+    }, 500),
+    [spaces]
+  );
+
+  // apply sizes updates from remote session
+  useEffect(() => {
+    const lastSizes = previousSizesRef.current;
+    const totalSize = currentTotalSize.current;
+    if (lastSizes && totalSize && sizes && lastSizes !== sizes) {
+      spaceRef.current?.resize(getAbsoluteSizes(sizes, totalSize));
+    }
+    previousSizesRef.current = sizes;
+  }, [sizes]);
 
   if (node.layout) {
     return (
       <SpaceContainer data-type="space-container">
-        <Allotment vertical={node.layout === Layout.Vertical}>
-          {node.children.map((space) => (
-            <Allotment.Pane key={space.id}>
-              <Space node={space} id={id} />
-            </Allotment.Pane>
-          ))}
+        <Allotment
+          vertical={node.layout === Layout.Vertical}
+          onReset={() => {
+            // todo: support more than two panel per space
```

Comment:
By default, Allotment will reset each space to `preferredSize` if provided. resizing it `[0,0]` when onReset is triggered is a trick to force allotment to resize space proportionally (default sizing at the very beginning) to identical width/height instead of `preferredSize`. If spaces are reset `preferredSize`, clicking on the border between two spaces would not have an effect.